packages/test/test-end-to-end-tests/src/test/summaries.spec.ts

```diff
@@ -307,6 +325,73 @@ describeNoCompat("Summaries", (getTestObjectProvider) => {
 		);
 	});
 
+	itExpects(
+		"full initialization of data object should not happen by default",
+		[
+			{
+				eventName: "fluid:telemetry:Container:Request_cancel",
+				error: "Non interactive/summarizer client's data object should not be initialized",
+			},
+		],
+		async () => {
+			const TestDataObjectType1 = "@fluid-example/test-dataStore1";
+			const dataStoreFactory1 = new DataObjectFactory(
+				TestDataObjectType1,
+				TestDataObject1,
+				[],
+				[],
+			);
+			const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+				[dataStoreFactory1.type, Promise.resolve(dataStoreFactory1)],
+			]);
+			const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+				dataStoreFactory1,
+				registryStoreEntries,
+				undefined,
+				[],
+			);
+
+			// Create a container for the first client.
+			const container1 = await provider.createContainer(runtimeFactory);
+			await assert.doesNotReject(
+				requestFluidObject<TestDataObject1>(container1, "/"),
+				"Initial creation of container and data store should succeed.",
+			);
+
+			// Create a summarizer for the container and do a summary shouldn't throw.
+			const createSummarizerResult = await createSummarizerFromFactory(
+				provider,
+				container1,
+				dataStoreFactory1,
+				undefined,
+				ContainerRuntimeFactoryWithDefaultDataStore,
+				registryStoreEntries,
+			);
+			const summarizer = createSummarizerResult.summarizer;
+			assert.doesNotThrow(
+				async () => summarizeNow(summarizer, "test"),
+				"Summarizing should not throw",
+			);
+
+			// In summarizer, load the data store should fail.
+			await assert.rejects(
+				async () =>
+					requestFluidObject<TestDataObject1>(createSummarizerResult.container, "/"),
+				(e) =>
+					e.message ===
+					"Non interactive/summarizer client's data object should not be initialized",
+				"This validates that a summarizer client doesn't fully initialize the data object.",
```

Comment:
```suggestion
				"Loading data store in summarizer did not throw as it should, or threw an unexpected error.",
```
Since this text is the error message we'll get if `requestFluidObject` does *not* throw as expected (or if it throws an error that is not the one we expect) I think this would be more descriptive to the person seeing the test fail.